### PR TITLE
Shrav dev2021

### LINKED ISSUE #1: Remove withBusyIndicatorServer and its calls
Code was borrowed from [daattali/advanced-shiny](https://github.com/daattali/advanced-shiny/blob/master/busy-indicator/helpers.R#L42) but is not needed. The shinyjs package can also be removed when this goes.

### LINKED ISSUE #6: Missing citation
On the Overview page, the "Cite this app as:" is missing.

### PR DESCRIPTION
**Pull request by shravanisamala**
I am just not sure if I should delete lines 896 - 913 and 519 - 565 because they are all commented out.

### CODE REVIEW

**Comment by rpc5102** [comment]
Lines 519 - 565 can be removed. Let's leave the other commented ones in for now, I will convert them to the new system when you're finished working on this app.

**Comment by shravanisamala** [comment]
I fixed the hints not going away after restarting the game as well as the description for the Normal Distribution that Dr. Pearl sent me. Thank you!

File: app.R [hunk 1]
```diff
@@ -31,14 +35,13 @@ ui <- list(
     sidebarMenu(
       id = "pages",
       menuItem("Overview", tabName = "overview", icon = icon("tachometer-alt")),
-      menuItem("Prerequisites", tabName = "concepts", icon = icon("book")),
-      menuItem("Game", tabName = "test", icon = icon("gamepad")),
+      menuItem("Prerequisites", tabName = "prerequisites", icon = icon("book")),
+      menuItem("Game", tabName = "game", icon = icon("gamepad")),
       menuItem("References", tabName = "references", icon = icon("leanpub"))
```
Comment (lbednarczyk) [inline app.R]: Here I think it would make the code more readable if each argument was on a new line.

File: DESCRIPTION [hunk 2]
```diff
@@ -4,6 +4,7 @@ Date: 2020-12-08
 Lifecycle: experimental
 Authors@R: c(
     person(given = "Yiyang", family = "Wang", role = c("aut", "cre"))
```
Comment (lbednarczyk) [inline DESCRIPTION]: I got an error for this line, there's a comma missing between lines 6 and 7.

**Review by lbednarczyk — CHANGES_REQUESTED** [review]
Hi Shravani,
The app looks and runs very well overall, there were just a few small formatting issues that I left comments for. I also noticed that math jax code wasn't rendering right when it was in a hint. One hint I saw looked like this: " \(\Phi\) is the cdf of the standard Normal". Other than that, everything else looked really good.

File: app.R (hunk 1 above)
Comment (shravanisamala) [reply app.R]: I left this as is because this is how it is on the app template

**Review by kqp5476 — DISMISSED** [review]
Screen Shot 2021-07-09 at 4 42 17 PM


I found no errors while exploring the app or playing the game a few times. However, the multiple-choice options are a bit interesting. I provided a screenshot explaining my situation and I will explain...
I was playing the game and I got some questions correct, and some incorrect. I answered the final question of my game incorrectly (hence why I died and had to restart). When I selected the answer choice, the red "cross" image and the pop-up message appeared like they are supposed to. So I left the pop-up message, and instead of clicking "restart" like the pop-up message instructs me to, I went back to the answer choices. Even though I got the question wrong the first time, I can keep selecting answer choices. Additionally, when I finally selected the correct answer choice, the green "check" image appeared (as shown in the attached screenshot). I'm not sure if this is intentional. But since the app works regardless of whether this is intentional, I do approve :)

**Review by lbednarczyk — DISMISSED** [review]
Hi Shravani,
Everything looks good now, the only thing I noticed is that the last updated statement still needs to be updated. Other than that everything ran smoothly so I'm approving it.

**Comment by shravanisamala** [comment]
@lbednarczyk Thanks, will fix that right now!

**Comment by shravanisamala** [comment]
@lbednarczyk @kqp5476 The update statement is updated now! Also, Kellien, for the multiple choice I did not notice that before! I could probably change it if Dr. Pearl and Hatfield want that but now that I think about it, maybe the person who created intended for it to be like this because in the csv file there was no column for feedback. Maybe this is their way of allowing the user to see the right answer? Or, it could have just been something they forgot to look into.

**Comment by neilhatfield** [comment]
@shravanisamala I've pushed some rather substantial changes (https://github.com/EducationShinyAppTeam/Probability_Applications/pull/5/commits/e86fa1b71dbf849cb9141f8212c31cd2e4a53c9d) to the app. I need you to do the following:

1. Pull my updated code to your computer and publish to the TLT server.
2. Go through the updated code and make sure there are no errors.
2. Thoroughly test out the play by playing the app; be sure that you play from several devices/browsers at the same time to check whether Player A interferes with Player B or not.
3. Carefully examine the questions and answers in the question bank. I came across a few typos in the hints, questions, etc. Additionally, do your best to check that there is in fact a correct answer for each question. Draw upon your fellow students for this as well.

**Comment by shravanisamala** [comment]
@neilhatfield Hi Dr. Hatfield, 
I just pushed the code to the TLT server. I went through the app and fixed all of the typos I found. I also played it simultaneously on two browsers (Google Chrome and Microsoft Edge) and the game still went smoothly. As for the correct answers part, do you mean that you think some questions' answer choices don't make sense?

**Comment by neilhatfield** [comment]
>  do you mean that you think some questions' answer choices don't make sense?

@shravanisamala I believe some of the wrong answers reflect common problematic ways of thinking for the given prompt. What I mean is that of the four answer choices, let's make sure that a valid answer is listed AND that answer also appears in the answer column. I don't remember which problem it was exactly (but it involved weights of 4.0, 4.5, 4.3) but the question asked about a weight value that didn't not actually appear in any of the answer expressions.

**Review by pophillips — CHANGES_REQUESTED** [review]
RECOMMENDED PRIORITY: URGENT
- On the prereq page, add an equivalent course background which would be indicate necessary knowledge to do well with this app. (ie. STAT 414 etc.) Maybe offer a link to a textbook or video if someone wanted to extend/supplement their learning.
- User should not be able to continue to next question without answering. I am able to click "next question" until I get to an easy question I might like, which would hinder my learning. I believe what would need to be added is an if statement requiring an answer to be submitted before allowing the next question button to be clicked.

**Review by Sburke26 — CHANGES_REQUESTED** [review]
Here are some things I took notice of while reviewing the app:

The screenshot.png is no in the doc folder.
The lifecycle in the description file does not match the badge in the README. 
The Heading 'App Description' is not present.
There is no  'Cite this app as...' entry.
Some object names are not the best representations of their purposes. Ex: "go" for the prereq page might be better as "goToPrereq".
In the console, "The `name` provided ('square-o') does not correspond to a known icon" appears numerous times.

A less important observation:

I notice the game allows you to hit the next button without answering the current question. It acts as sort of a "Skip" button. If that is its intention, it is ok as is. However, it may be beneficial to prevent this. 





Maybe dont allow next question without aswering currwent

**Review by robchapp — CHANGES_REQUESTED** [review]
The app runs well, and has a defined purpose however there are some minor tweaks to be done:

- There is no cite app
- The dates do not match on the description and readme 
- The lifecycles do not match on description and readme
- There is no screenshot
- The heading for app description in the readme is missing
- The icon for the buttons in the game do not exist and create numerous errors in the console

**Review by robchapp — CHANGES_REQUESTED** [review]
App is looking good, just small things like acknowledgments and descripiton/readme files need fixed. Don't forget to add yourself to the files

**Comment by Sburke26** [comment]
Thank you for the review.
I made some further adjustments to fix issues.
Let me know if you see anything else.

**Comment by neilhatfield** [comment]
@Sburke26 
I've made some minor formatting modifications and have gone through the code. Please go through the [Issues](https://github.com/EducationShinyAppTeam/Probability_Applications/issues) and see if any of them can be closed. If they can, list them in the Development list at the bottom of the right-hand column of this Pull Request. Once that is done, let me know and I'll complete the Pull Request and deploy the updated version.

**Comment by Sburke26** [comment]
@neilhatfield
 I listed the two issues that can be resolved with the pull request, so it should be good to go.

**Pull request merged**